functions/src/endpoint_paths/event.ts

```diff
@@ -0,0 +1,128 @@
+        || !req.body.date || req.body.date === "" || !req.body.location || req.body.location === ""
+        || !req.body.points || req.body.points === "" || !req.body.point_type_id || req.body.point_type_id === ""
+        || !req.body.point_type_name || req.body.point_type_name === "" || !req.body.point_type_description
+        ||req.body.point_type_description === "" || !req.body.house || req.body.house === "") {
+            if (!req.body) {
+                console.error("Missing Body")
+            }
+            else if (!req.body.name || req.body.name === "") {
+                console.error("Missing name")
+            }
+            else if (!req.body.details || req.body.details === "") {
+                console.error("Missing details")
+            }
+            else if (!req.body.date || req.body.date === "") {
+                console.error("Missing date")
+            }
+            else if (!req.body.location || req.body.location === "") {
+                console.error("Missing location")
+            }
+            else if (!req.body.points || req.body.points === "") {
+                console.error("Missing points")
+            }
+            else if (!req.body.point_type_id || req.body.point_type_id === "") {
+                console.error("Missing point_type_id")
+            }
+            else if (!req.body.point_type_name || req.body.point_type_name === "") {
+                console.error("Missing point_type_name")
+            }
+            else if (!req.body.point_type_description || req.body.point_type_description === "") {
+                console.error("Missing point_type_description")
+            }
+            else if (!req.body.house || req.body.house === "") {
+                console.error("Missing house")
+            }
+            const error = APIResponse.MissingRequiredParameters()
+            res.status(error.code).send(error.toJson())
+    } else {
+        try {
+            let user = await getUser(req["user"]["user_id"])
+            if (user.permissionLevel === UserPermissionLevel.RESIDENT) {
+                const error = APIResponse.InvalidPermissionLevel()
+                res.status(error.code).send(error.toJson())
+                return
+            }
+            const event_date = new Date(req.body.date)
+            let today = new Date()
+            // Must set today to earliest possible time so event_date set to today's date will not error
+            today.setHours(0,0,0,0)
+            if (event_date < today) {
+                const error = APIResponse.DateNotInRange()
+                res.status(error.code).send(error.toJson())
+            }
+            const didAddEvent = await addEvent(req.body.name, req.body.details, event_date,
+                req.body.location, parseInt(req.body.points), req.body.point_type_id, req.body.point_type_name,
+                req.body.point_type_description, req.body.house, user.id)
+            const success = APIResponse.Success()
```

Comment:
As I mention below, you should not pass the individual parts of point type. Call the getPointTypeById and pass the returned model into this function